modules/lang-mustache/src/internalClusterTest/java/org/elasticsearch/script/mustache/MultiSearchTemplateIT.java

```diff
@@ -200,9 +199,10 @@ public void testCCSCheckCompatibility() throws Exception {
         Exception ex = response.getFailure();
         assertThat(ex.getMessage(), containsString("[class org.elasticsearch.action.search.SearchRequest] is not compatible with version"));
         assertThat(ex.getMessage(), containsString("'search.check_ccs_compatibility' setting is enabled."));
-        assertEquals(
-            "This query isn't serializable with transport versions before " + TransportVersion.current(),
-            ex.getCause().getMessage()
-        );
+
+        String expectedCause = "[fail_before_current_version] was released first in version XXXXXXX, failed compatibility check trying to"
```

Comment:
Can you get the failing "future" version from the FailBeforeCurrentVersionQuery test query builder? I guess getting the failing node version is more difficult, so maybe no way to get around the string replace operation here? Would be nice to avoid it but not a blocker for me.

Reply:
I added this in a new commit.

Doing that forced me to realize there is one minor problem here.

The error message has the format:

> [<query_type>] was released first in version X, failed compatibility check trying to send it to node with version Y

In production settings, the version "Y" will be correct - the actual TransportVersion being used.

But for the check_ccs_compatibility=true condition, it won't actually report current TransportVersion - it will report TransportVersion.MINIMUM_CCS_VERSION, because that's what we pass into the TransportSearchHelper check method here: https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/search/TransportSearchHelper.java#L108

It is correct to pass in TransportVersion.MINIMUM_CCS_VERSION to do the minimal check, but the error message is then wrong - we aren't checking current version, we are checking the previous release version. So that error message could be confusing to kibana testers.

But we can't change the error message to say "minimal version" because in production it reports actual version.

So I just left it as is.